injector/injector.go

```diff
@@ -310,3 +226,60 @@ func (ti *TrackerInjector) InjectTracker(vastXML string, NURL string) string {
 	}
 	return outputXML.String()
 }
+
+func (ti *TrackerInjector) addTrackingEvent(outputXML *strings.Builder, creativeId string) {
```

Comment:
I wonder if these functions can be optimized even further from an amount of code standpoint since i still see a lot of common code.

What do you think of something like this? Would this work? 

```
func (ti *TrackerInjector) writeTrackingEvent(outputXML *strings.Builder, urls []string, tagName string, creativeId string, eventType string) {
    if creativeId != "" && eventType != "" {
        ti.provider.PopulateEventMacros(creativeId, eventType, "")
    }
    for _, url := range urls {
        outputXML.WriteString("<" + tagName + "><![CDATA[")
        ti.replacer.Replace(outputXML, url, ti.provider)
        outputXML.WriteString("]]></" + tagName + ">")
    }
}

// Application for example
func (ti *TrackerInjector) addClickTrackingEvent(outputXML *strings.Builder, creativeId string) {
    ti.writeTrackingEvent(outputXML, ti.events.VideoClicks, "ClickTracking", creativeId, "")
}
```

Reply:
Done.